climada/hazard/base.py

```diff
@@ -760,6 +775,34 @@ def select(self, event_names=None, date=None, orig=None, reg_id=None, reset_freq
         haz.sanitize_event_ids()
         return haz
 
+    def select_tight_cent(self, buffer=0.0):
+        """
+        Reduce hazard to centroids to minimal box containing all non-zero
+        intensity points.
+
+        Parameters
+        ----------
+        buffer : float, optional
+            Buffer of box in degrees. The default is 0.0.
+
+        Returns
+        -------
+        Hazard
+            Copy of the Hazard with centroids reduced to minimal box. All other
+            hazard properties are carried over without changes.
+
+        See also
+        --------
+        self.select: Method to select centroids by lat/lon extent
+
+        """
+
+        cent_nz = self.intensity.nonzero()[1]
```

Comment:
This will create a lot of duplicate indices which creates unnecessary overhead in the call to `latlon_bounds` below. I suggest to reduce the intensity matrix to a single row before extracting non-zero columns:
```suggestion
        cent_nz = (self.intensity != 0).sum(axis=0).nonzero()[1]
```

Reply:
Good point. Not sure which one is faster, but I think your suggestion is.

Reply:
Done.